enhancements/microshift/microshift-router-configuration.md

```diff
@@ -0,0 +1,476 @@
+    - <hostname>
+    interfaces:
+    - <NIC name>
+    ipAddresses:
+    - <IP address>
+```
+
+### Risks and Mitigations
+Some of the features in this enhancement proposal rely on the
+[LoadBalancer controller](loadbalancer-service-support.md).
+Some of the features rely on NetworkPolicy resources, so the CNI needs
+to support that.
+
+### Drawbacks
+Some of the features depend on a non-agnostic CNI design, which is the
+`LoadBalancer` controller. A different behavior should be expected if using a
+different CNI.
+
+## Design Details
+#### Why LoadBalancer service
+In the current implementation router ports are fixed and can not be configured.
+Back when there was no support for `LoadBalancer` service types, router was
+forced to use a different way of getting exposed. Using ports 80 and 443 meant
+that NodePort service types can not be used. Using host network would also
+bind port 1936 to the host, which is used for internal metrics. The only option
+that was left was using host ports.
+
+Ports 80 and 443 are configured to use `hostPort` option. This will instruct
+CRI to bind container ports 80 and 443 to the host in the form of iptables
+rules, forwarding incoming traffic to the pod's IP address.
+
+The following limitations apply to the current setup:
+* Iptables rules bypass firewalld settings. This means any firewall
+  configuration applying to host ports will be ignored. It is not possible
+  to control incoming traffic to these ports.
+* NetworkPolicy resources do not apply to host ports. These resources are
+  able to control how traffic reaches which endpoints, including external
+  sources.
+* Any pod using this option is unable to scale within the same node.
+* Port numbers are bound to the deployment, therefore any changes require
+  a restart of the pod.
+
+##### Using LoadBalancer service type
+Common ways of exposing services outside of the cluster are using services of
+types NodePort or LoadBalancer. In MicroShift, ports from the service are
+picked up from ovnk (the CNI) and turned into iptables rules to forward them
+to the service IP. In this case, default router should be exposed using a
+LoadBalancer service type.
+
+ When using this kind of service ovnk will configure special iptables rules
+ that will forward traffic to the service IP instead. These rules are also
+ configured to use the node IP as destination address, as per
+ [LoadBalancer support](loadbalancer-service-support.md) feature.
+
+#### Firewalling ports
+Using LoadBalancer service will create special iptables rules with more
+precedence than those from firewalld. This means a `LoadBalancer` service is
+immune to such configurations. In order to firewall ports the user needs to
+take action by either creating `NetworkPolicy` resources or disabling the
+router.
+
+Disabling the router might be disruptive, as not all traffic may come from
+external sources. This procedure also requires a restart, which might not be
+desirable/possible in all situations.
```

Comment:
This would be worth calling out in Risks/Mitigations, since it's risk this EP introduces.

Reply:
Done